### PR TITLE
Get rid of shuttle auth spam

### PR DESCRIPTION
**Pull request by ghost**
:cl: steamp0rt
tweak: Assholes can no longer rape both ears and the chat by spamming the shuttle authorization
/:cl:

### CODE REVIEW

**Comment by yobob591** [comment]
-1 my ears enjoy being violated

**Comment by ghost** [comment]
yeah it's cool - 1 classic

**Comment by Manly-Dwarf** [comment]
But where will I get my excuses to adminbus nerds when you remove this?!

**Comment by Carbonhell** [comment]
Fixes https://github.com/HippieStation/HippieStation/issues/6466, add it to the desc to autoclose it if it's merged
anyway yes +1, fuck this shit

File: code/modules/shuttle/emergency.dm [hunk 1]
```diff
@@ -10,6 +10,7 @@
 	icon_keyboard = "tech_key"
 	var/auth_need = 3
 	var/list/authorized = list()
+	var/list/last_action = list() // hippie -- kill shuttle auth spam
```
Comment (Carbonhell) [inline code/modules/shuttle/emergency.dm]: modularize?

File: code/modules/shuttle/emergency.dm [hunk 2]
```diff
@@ -68,7 +69,11 @@
 		return
 
 	var/old_len = authorized.len
-
+	
+	if(LAZYLEN(last_action) && last_action[user] && last_action[user] >= world.time + 10) // hippie start -- stop shuttle auth spam
```
Comment (Carbonhell) [inline code/modules/shuttle/emergency.dm]: can't you modularize this easily aswell?

**Review by Carbonhell — DISMISSED** [review]

**Comment by ghost** [comment]
Modularized

**Comment by HippieBartels** [comment]
+1

**Comment by Eagle-Eyess** [comment]
+1

File: code/modules/shuttle/emergency.dm [hunk 3]
```diff
@@ -69,6 +69,8 @@
 
 	var/old_len = authorized.len
 
+	HIPPIE_HOOK_SHUTTLE_AUTH
```
Comment (ghost) [inline code/modules/shuttle/emergency.dm]: needs a comment to show this is a hippie line with a search and to explain why

**Review by ghost — CHANGES_REQUESTED** [review]

**Review by ghost — DISMISSED** [review]

File: code/__DEFINES/~hippie_defines/hooks.dm [hunk 4]
```diff
@@ -0,0 +1 @@
+#define HIPPIE_HOOK_SHUTTLE_AUTH if(LAZYLEN(last_action) && last_action[user] && last_action[user] >= world.time + 10) return; LAZYSET(last_action, user, world.time)
```
Comment (ghost) [inline code/__DEFINES/~hippie_defines/hooks.dm:1]: world.time >= 10 SECONDS (i assume?)
i'd rather this define was unambigious about the time delay

**Review by ghost — CHANGES_REQUESTED** [review]

**Review by ghost — APPROVED** [review]

File: code/__DEFINES/~hippie_defines/hooks.dm (hunk 4 above)
Comment (ghost) [reply code/__DEFINES/~hippie_defines/hooks.dm:1]: 1 second

Comment (ghost) [reply code/__DEFINES/~hippie_defines/hooks.dm:1]: yea im being dumb

**Pull request merged**